invenio_records_resources/services/files/generators.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 CERN.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""File permissions generators."""
+
+
+from invenio_access.permissions import any_user, system_process
+from invenio_records_permissions.generators import Generator
+from invenio_search.engine import dsl
+
+
+# TODO: Naming
+class AnyUserIfFileIsLocal(Generator):
```

Comment:
In general, we need (and will more and more) permissions over more fine grained file actions. However, I'm not convinced about making the generator aware of `S` and `E` (Storage classes enum). On the other hand, I'm not sure how to abstract it except having "can_<action>" functions in the `Transfer` object... (seems like a permission duplicate).